tree-sitter-darklang/grammar.js

```diff
@@ -854,6 +855,24 @@ module.exports = grammar({
         field("symbol_close_angle", alias(">", $.symbol)),
       ),
 
+    //
+    // Function type reference
+    //  'a -> 'b -> 'c
+    fn_type_reference: $ =>
+      prec.right(
+        seq(
+          $.type_reference,
+          repeat1(
+            prec.left(
+              seq(
+                field("symbol_arrow", alias("->", $.symbol)),
+                $.type_reference,
+              ),
+            ),
+          ),
+        ),
+      ),
```

Comment:
This might fit more naturally like this: 
```suggestion
    fn_type_reference: $ =>
      prec.right(
        seq(
          repeat1(
            prec.left(
              seq(
                $.type_reference,
               field("symbol_arrow", alias("->", $.symbol)),
              ),
            ),
          ),
          $.type_reference,
        ),
      ),
```

Feel free to disregard or leave for another PR, though